targets/actionscript/source/com/playfab/PlayFabSettings.as.ejs

```diff
@@ -18,12 +25,48 @@ package com.playfab
         public static const AD_TYPE_IDFA:String = "Idfa";
         public static const AD_TYPE_ANDROID_ID:String = "Adid";
 <% } %>
-        public static function GetURL():String
+        public static function GetURL(urlPath:String):String
         {
             var baseUrl:String = ProductionEnvironmentURL;
-            if(baseUrl.indexOf("http") == 0)
-                return baseUrl;
-            return "https://"+TitleId+baseUrl;
+            var fullUrl:String;
+            if(baseUrl.indexOf("http") != 0)
+            {
+                if(VerticalName != null)
+                {
+                    fullUrl = "https://" + VerticalName;
+                }
+                else
+                {
+                    fullUrl = "https://" + TitleId;
+                }
+
+                fullUrl = fullUrl + baseUrl + urlPath;
+            }
+            else
+            {
+                fullUrl = baseUrl + urlPath;
+            }
+
+            var getParams:Dictionary = RequestGetParams;
+            var firstParam:Boolean = true;
+            for(var key:String in getParams)
+            {
+                var value:String = getParams[key];
+
+                if(firstParam)
+                {
+                    fullUrl += "?";
+                    firstParam = false;
+                }
+                else
+                {
+                    fullUrl += "&";
+                }
+
+                fullUrl = fullUrl + key + "=" + value;
+            }
+
+            return fullUrl;
```

Comment:
I think we should make a tech-debt task for a future sprint to optimize this string construction.
Some web results seem to indicate the best would be an array of strings, with a join at the end, much like Node/JS.
But this should be committed now, and published today.

Reply:
Makes sense, I created a task for that